src/contracts/fund/participation/Participation.ts

```diff
@@ -285,6 +302,55 @@ export class Participation extends Contract {
       amgu,
     });
   }
+
+  /**
+   * Redeem all shares
+   *
+   * @param from The address of the sender.
+   */
+  public redeem(from: Address) {
+    const validate = async () => {
+      const shares = new Shares(this.environment, (await this.getRoutes()).shares);
+      const ownedShares = await shares.getBalanceOf(from);
+      if (ownedShares.isZero()) {
+        throw new NoSharesToRedeemError(from);
+      }
+    };
+
+    return this.createTransaction({
+      from,
+      method: 'redeem',
+      args: undefined,
+      validate,
+    });
+  }
+
+  /**
+   * Redeem a quantity of shares
+   *
+   * @param from The address of the sender.
+   * @param shareQuantity The quantity of shares to redeem
+   */
+  public redeemQuantity(from: Address, shareQuantity: BigNumber) {
+    const validate = async () => {
+      const shares = new Shares(this.environment, (await this.getRoutes()).shares);
+      const ownedShares = await shares.getBalanceOf(from);
+      if (ownedShares.isZero()) {
+        throw new NoSharesToRedeemError(from);
+      }
+
+      if (ownedShares.isLessThan(shareQuantity)) {
+        throw new NotEnoughSharesToRedeemError(from);
```

Comment:
Add the ownedShares quantity to the error as a class variable of the error class.